Makefile

```diff
@@ -168,15 +168,15 @@ generate.init: $(TERRAFORM_PROVIDER_SCHEMA) pull-docs
 # Test utilities
 # TODO(muvaf): Move most of this to build submodule.
 
-uptest: $(KIND) $(KUBECTL) $(HELM3) $(UP) $(KUTTL)
+uptest: $(KIND) $(KUBECTL) $(HELM3) $(UP) $(KUTTL) $(UPTEST)
 	@$(INFO) running uptest using kind $(KIND_VERSION)
 	@./cluster/install_provider.sh || $(FAIL)
 	@echo "$${UPTEST_EXAMPLE_VALUE_REPLACEMENTS}" > $(WORK_DIR)/replacements.yaml
-	@KIND=$(KIND) KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) go run github.com/upbound/official-providers/testing/cmd --data-source "$(WORK_DIR)/replacements.yaml" || $(FAIL)
+	@KIND=$(KIND) KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e ${EXAMPLE_LIST} --default-conditions="Test" --data-source "$(WORK_DIR)/replacements.yaml" || $(FAIL)
 
-uptest-local: $(KUBECTL) $(KUTTL)
+uptest-local: $(KUBECTL) $(KUTTL) $(UPTEST)
 	@$(INFO) running automated tests with uptest using current kubeconfig $(KIND_VERSION)
-	@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) go run github.com/upbound/official-providers/testing/cmd || $(FAIL)
+	@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e ${EXAMPLE_LIST} --default-conditions="Test" --data-source "$(WORK_DIR)/replacements.yaml" || $(FAIL)
 
 cluster_dump: $(KUBECTL)
 	@mkdir -p ${DUMP_DIRECTORY}
```

Comment:
> @cat /tmp/automated-tests/case/*.yaml > ${DUMP_DIRECTORY}/kuttl-inputs.yaml

By using os.tmpdir, we no longer write to `/tmp/automated-tests`. I believe we should better support taking that directory as input in `uptest` makes sense.

Reply:
Opened https://github.com/upbound/uptest/pull/29

Reply:
Backporting so that I can cut a patch with that: https://github.com/upbound/uptest/pull/30

Reply:
@turkenh Thanks for the quick fix! Is this error familiar?
```
 Running kuttl tests at 
2022/10/18 15:00:12 running without a 'kuttl-test.yaml' configuration
Error: no test directories provided, please provide either --config or test directories on the command line
uptest-v0.1.1: error: cannot run e2e tests successfully: cannot execute tests: kuttl failed: exit status 255
15:00:12 [FAIL]
make: *** [Makefile:176: uptest] Error 1
Error: Process completed with exit code 2.
```

Reply:
> Running kuttl tests at 

Above line looks suspicious. Not sure how it can be empty but you can try passing the new flag to set test dir.

Reply:
I have added `--test-directory="${DUMP_DIRECTORY}"` and it is set here https://github.com/upbound/provider-aws/blob/main/.github/workflows/uptest.yml#L167

Reply:
@muvaf https://github.com/upbound/provider-aws/pull/82/commits/6b42e4f9fe2d0e86a969261a27a5015bb3fbcf25 was missing, but still didn't work since workflows used from main 🤦 So, we need to merge to see.